pallets/rewards/src/lib.rs

```diff
@@ -517,6 +543,51 @@ pub mod pallet {
             T::AccountManager::get_settlements_by_role(CircuitRole::Executor)
         }
 
+        pub fn author() -> Option<T::AccountId> {
+            let digest = <frame_system::Pallet<T>>::digest();
+            let pre_runtime_digests = digest.logs.iter().filter_map(|d| d.as_pre_runtime());
+            T::FindAuthor::find_author(pre_runtime_digests)
+        }
+
+        pub fn process_author() -> bool {
+            if let Some(author) = Self::author() {
+                AuthorsThisPeriod::<T>::mutate(|authors| {
+                    *authors.entry(author).or_insert(0) += 1;
+                    // If we have more authors than MAX_AUTHORS, remove the author with the least blocks
+                    if authors.len() > (MAX_AUTHORS as usize) {
+                        let min_count = authors.values().min().cloned().unwrap_or_default();
+                        let least_blocks_authors: Vec<T::AccountId> = authors
+                            .iter()
+                            .filter_map(|(author, count)| {
+                                if *count == min_count {
+                                    Some(author.clone())
+                                } else {
+                                    None
+                                }
+                            })
+                            .collect();
+                        for author in least_blocks_authors {
```

Comment:
i think you can just use sort_by here

Reply:
changed to use min_by https://github.com/t3rn/t3rn/pull/966/commits/000692d75a8ef05f8a929416721f25067b12ca76